tools/circleci/requirements.txt

```
@@ -0,0 +1,3 @@
+pip>=20.0
```

Optional: When `pip` upgrades it's usually a very breaking change, so I'm tempted to be more conservative on upper bounds, but on the other hand at least if it does break we know we have to patch it immediately. Up to you.
```suggestion
pip>=20.0,<22.0
```

---

This is a good thought, but I'm going to leave this just so that it's consistent with various other places where we also don't set an upper bound for `pip`. So if there is a breaking change then everything will break anyway.